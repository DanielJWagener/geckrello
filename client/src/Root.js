// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
