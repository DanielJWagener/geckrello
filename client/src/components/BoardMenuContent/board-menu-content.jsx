import React from "react";

import Archive from "../Archive/archive.component";
import ChangeBackground from "../ChangeBackground/change-background.component";
import SidebarPanel from "../BoardMenuPanel/board-menu-panel.component";
import SidebarItem from "../BoardMenuItem/board-menu-item";

class SidebarMenu extends React.Component {
  state = { panel: "none" };

  togglePanel = panel => {
    this.setState({ panel });
  };

  render() {
    return (
      <div className="sidebar-menu">
        <h2 className="sidebar-menu__heading">Menu</h2>
        <div className="sidebar-menu__close" onClick={this.props.toggleSidebar}>
          &times;
        </div>
        <hr />
        <SidebarItem toggleTarget="Archive" togglePanel={this.togglePanel}>
          Show Archive
        </SidebarItem>
        <SidebarItem
          toggleTarget="ChangeBackground"
          togglePanel={this.togglePanel}
        >
          Change Background
        </SidebarItem>
        <SidebarItem
          toggleTarget="BoardDescription"
          togglePanel={this.togglePanel}
        >
          Edit Board Description
        </SidebarItem>
        <SidebarItem toggleTarget="Friends" togglePanel={this.togglePanel}>
          Manage Friends
        </SidebarItem>

        <SidebarPanel
          visible={this.state.panel === "Archive"}
          togglePanel={this.togglePanel}
          heading="Archive"
        >
          <Archive />
        </SidebarPanel>
        <SidebarPanel
          visible={this.state.panel === "ChangeBackground"}
          togglePanel={this.togglePanel}
          heading="Change Background"
        >
          <ChangeBackground />
        </SidebarPanel>
        <SidebarPanel
          visible={this.state.panel === "BoardDescription"}
          togglePanel={this.togglePanel}
          heading="Board Description"
        ></SidebarPanel>
        <SidebarPanel
          visible={this.state.panel === "Friends"}
          togglePanel={this.togglePanel}
          heading="Friends"
        ></SidebarPanel>
      </div>
    );
  }
}

export default SidebarMenu;
