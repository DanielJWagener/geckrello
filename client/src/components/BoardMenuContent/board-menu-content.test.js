import { shallow } from "enzyme";
import React from "react";
import BoardMenuContent from "./board-menu-content.component";
import Root from "../../Root";

it("renders BoardMenuContent component", () => {
  expect(shallow(<BoardMenuContent />).debug()).toMatchSnapshot();
});
