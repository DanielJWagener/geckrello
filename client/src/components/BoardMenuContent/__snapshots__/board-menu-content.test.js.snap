// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardMenuContent component 1`] = `
"<div className=\\"sidebar-menu\\">
  <h2 className=\\"sidebar-menu__heading\\">
    Menu
  </h2>
  <div className=\\"sidebar-menu__close\\" onClick={[undefined]}>
    Ã—
  </div>
  <hr />
  <BoardMenuItem toggleTarget=\\"Archive\\" togglePanel={[Function]}>
    Show Archive
  </BoardMenuItem>
  <BoardMenuItem toggleTarget=\\"ChangeBackground\\" togglePanel={[Function]}>
    Change Background
  </BoardMenuItem>
  <BoardMenuPanel visible={false} togglePanel={[Function]} heading=\\"Archive\\">
    <Connect(Archive) />
  </BoardMenuPanel>
  <BoardMenuPanel visible={false} togglePanel={[Function]} heading=\\"Change Background\\">
    <ChangeBackground />
  </BoardMenuPanel>
  <BoardMenuPanel visible={false} togglePanel={[Function]} heading=\\"Board Description\\" />
  <BoardMenuPanel visible={false} togglePanel={[Function]} heading=\\"Friends\\" />
</div>"
`;
